test/integration-tests/tests/jd_integration.rs

```diff
@@ -108,3 +114,130 @@ async fn jdc_does_not_stackoverflow_when_no_token() {
     tokio::time::sleep(Duration::from_secs(3)).await;
     assert!(tokio::net::TcpListener::bind(jdc_addr).await.is_err());
 }
+
+///The issue arises from the following message flow. JDC first receives a `NewTemplate`,
+/// prompting it to send a `RequestTransactionData` message to TP. TP responds with
+/// `RequestTransactionDataSuccess`, after which JDC sends a `DeclareMiningJob` to JDS. JDS then
+/// requests missing transactions from JDC via `ProvideMissingTransactions`. While JDC is in the
+/// process of responding with `ProvideMissingTransactionsSuccess`, a block is found, leading JDC
+/// to send a `SubmitSolution` to JDS. However, since JDS has not yet received the requested
+/// transactions from JDC, it results in shutdown. This test tries to mimic the given situation.
```

Comment:
I think this is too verbose. the test name is already very clear. if you still wanna add logs for this, please first use "//" rather than "///" and keep it simple and accurate.

Reply:
`///` is for `pub` modules and `//` is for internal docs